dataflow/gpu-workers/conftest.py

```diff
@@ -0,0 +1,503 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+from dataclasses import dataclass
+import itertools
+import json
+import logging
+import multiprocessing as mp
+import os
+import re
+import platform
+import subprocess
+import sys
+import time
+from typing import Any, Callable, Dict, Iterable, Optional
+import uuid
+
+import pytest
+
+# Default options.
+UUID = uuid.uuid4().hex[0:6]
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+REGION = "us-central1"
+
+RETRY_MAX_TIME = 5 * 60  # 5 minutes in seconds
+
+HYPHEN_NAME_RE = re.compile(r"[^\w\d-]+")
+UNDERSCORE_NAME_RE = re.compile(r"[^\w\d_]+")
+
+PYTHON_VERSION = "".join(platform.python_version_tuple()[0:2])
+
+
+@dataclass
+class Utils:
+    uuid: str = UUID
+    project: str = PROJECT
+    region: str = REGION
+
+    @staticmethod
+    def hyphen_name(name: str) -> str:
+        unique_name = f"{name}-py{PYTHON_VERSION}-{UUID}"
+        return HYPHEN_NAME_RE.sub("-", unique_name)
+
+    @staticmethod
+    def underscore_name(name: str) -> str:
+        return UNDERSCORE_NAME_RE.sub("_", Utils.hyphen_name(name))
+
+    @staticmethod
+    def storage_bucket(name: str) -> str:
+        from google.cloud import storage
+
+        storage_client = storage.Client()
+        bucket = storage_client.create_bucket(Utils.hyphen_name(name))
+
+        logging.info(f"Created storage_bucket: {bucket.name}")
+        yield bucket.name
+
+        # Print all the objects in the bucket before deleting for debugging.
```

Comment:
Does the cleanup code  after `yield`  get executed if we only create one bucket?

Reply:
Yes, fixtures do cleanup after the yield.